File: backend/components/metiers/administration_publique/utils/ai/fallback/ai_fallback.test.js
// ai_fallback.test.js – Test unitaire ultra avancé fallback IA (JS)
const { aiFallback } = require('./ai_fallback');

describe('aiFallback – Fallback IA (JS)', () => {
  it('doit retourner un fallback pour une entrée non traitée', () => {
    const res = aiFallback('input-non-traite');
    expect(res.status).toBe('FALLBACK');
    expect(res.data).toBe('input-non-traite');
    expect(res.audit).toBe(true);
  });

  it('doit retourner une erreur si input vide', () => {
    const res = aiFallback();
    expect(res.status).toBe('ERROR');
  });
});

File: backend/components/metiers/administration_publique/utils/ai/fallback/ai_fallback.test.py
# ai_fallback.test.py – Test unitaire ultra avancé fallback IA (Python)
from .ai_fallback import ai_fallback

def test_ai_fallback_ok():
    res = ai_fallback('input-non-traite')
    assert res['status'] == 'FALLBACK'
    assert res['data'] == 'input-non-traite'
    assert res['audit'] is True

def test_ai_fallback_error():
    res = ai_fallback(None)
    assert res['status'] == 'ERROR'
